### PR TITLE
IAP: Stub for the creating an order API

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #7515

### Description

Adds stubbed API call so errors can be tested/integrated

### Testing instructions

* Perform purchase via showcase activity
* Notice that network/server error pop-ups randomly

### CODE REVIEW

File: libs/iap/src/main/java/com/woocommerce/android/iap/internal/network/IAPMobilePayAPI.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    suspend fun createAndConfirmOrder(
+        remoteSiteId: Long,
+        productIdentifier: String,
+        price: Long,
+        currency: String,
+        purchaseToken: String,
+    ): CreateAndConfirmOrderResponse
+}
+
+@Suppress("MagicNumber")
```
Comment (AnirudhBhat) [inline libs/iap/src/main/java/com/woocommerce/android/iap/internal/network/IAPMobilePayAPI.kt:22]: 🔍 ❓ nitpick: I believe this will go away once we have the backend ready. Do you think it would help us to leave a todo note here to remove this down the line?

**Review by AnirudhBhat — APPROVED** [review]
LGTM :shipit:

File: libs/iap/src/main/java/com/woocommerce/android/iap/internal/network/IAPMobilePayAPI.kt (hunk 1 above)
Comment (kidinov) [reply libs/iap/src/main/java/com/woocommerce/android/iap/internal/network/IAPMobilePayAPI.kt:22]: I believe TODOs are worst than having tickets in the project, that's why I added [a ticket] =)(https://github.com/woocommerce/woocommerce-android/issues/7563)

Good good especially about "single, shared, visible backlog" from [the article](https://www.osedea.com/en/blog/should-you-put-todos-in-the-source-code):

> TL;DR: Use temporary TODOs as you work on a feature, but make sure to treat them before merging. Either add a link to a proper issue in your TODO, or remove it from the code. You should have a single, shared, visible backlog of things to do. Don't let TODOs become an invisible one.
